theano/sandbox/gpuarray/blas.py

```diff
@@ -331,6 +331,92 @@ def c_code_cache_version(self):
 
 gpu_dot22 = GpuDot22()
 
+class GpuGemmBatch(BlasOp):
+    __props__ = ('inplace',)
+
+    def __init__(self, inplace=False):
+        self.inplace = inplace
+        if self.inplace:
+            self.destroy_map = {0: [0]}
+
+    def make_node(self, C, alpha, A, B, beta):
+        ctx_name = infer_context_name(C, A, B)
+        A = as_gpuarray_variable(A, ctx_name)
+        B = as_gpuarray_variable(B, ctx_name)
+        C = as_gpuarray_variable(C, ctx_name)
+        alpha = as_tensor_variable(alpha)
+        beta = as_tensor_variable(beta)
+        assert alpha.ndim == 0
+        assert beta.ndim == 0
+        assert A.ndim == 3
+        assert B.ndim == 3
+        assert C.ndim == 3
+        assert A.dtype == B.dtype == C.dtype
+        return Apply(self, [C, alpha, A, B, beta], [C.type()])
+
+    def perform(self, node, inputs, outputs):
+        C, alpha, A, B, beta = inputs
+        if self.inplace and C.flags.forc:
+            C *= beta
+            C += alpha * blas.batched_dot(A, B)
+            outputs[0][0] = C
+        else:
+            outputs[0][0] = alpha * blas.batched_dot(A, B) + beta * C
```

Comment:
Where do you take you batched_dot implementation?  I didn't see anything in the libgpuarray PR.

You can just remove the perform if it doesn't work.